src/core/tasks_pool.c

```diff
@@ -0,0 +1,137 @@
+#include "tasks_pool.h"
+
+xccl_status_t tasks_pool_init(xccl_progress_queue_t* handle) {
+    handle->ctx = (void*) malloc(sizeof(xccl_tasks_pool_t));
+    xccl_tasks_pool_t* ctx = (xccl_tasks_pool_t*) handle->ctx;
+    ctx->deque_size = (xccl_lib_global_config.tasks_pool_size / LINE_SIZE) + 1;
+    ctx->tasks = (ucc_coll_task_t ***) calloc(2 * ctx->deque_size, sizeof(ucc_coll_task_t **));
+    if (ctx->tasks == NULL) {
+        return XCCL_ERR_NO_MEMORY;
+    }
+    ctx->tasks[0] = (ucc_coll_task_t **) calloc(LINE_SIZE, sizeof(ucc_coll_task_t *));
+    if (ctx->tasks[0] == NULL) {
+        return XCCL_ERR_NO_MEMORY;
+    }
+    ctx->tasks[ctx->deque_size] = (ucc_coll_task_t **) calloc(LINE_SIZE, sizeof(ucc_coll_task_t *));
+    if (ctx->tasks[ctx->deque_size] == NULL) {
+        return XCCL_ERR_NO_MEMORY;
+    }
+    ucs_spinlock_init(&(ctx->increase_pool_locks[0]), 0);
+    ucs_spinlock_init(&(ctx->increase_pool_locks[1]), 0);
+    ucs_spinlock_init(&(ctx->linked_list_lock), 0);
+    ctx->linked_list_head              = NULL;
```

Comment:
i think we should use ucs_list_link_t and the corresponding API from ucs/datastruct/list.h: ucs_list_add_head/tail, etc.

Reply:
Done